### PR TITLE
ta-lib: init at 0.4.0

### PR DESCRIPTION
**Pull request by rafa-dot-el**
Added TA-Lib, a library used for technical analysis.

Homepage: https://ta-lib.org/

This library is a mandatory dependency for financial and trading tools, for example:
 - https://github.com/twopirllc/pandas-ta
 - https://mrjbq7.github.io/ta-lib/

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/ta-lib/default.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  name = "ta-lib";
+  version = "0.4.0";
+  src = fetchFromGitHub {
+    owner = "rafa-dot-el";
+    repo = "talib";
+    rev = "${version}";
+    sha256 = "sha256-bIzN8f9ZiOLaVzGAXcZUHUh/v9z1U+zY+MnyjJr1lSw=";
+  };
+
+  nativeBuildInputs = [ gcc pkg-config autoreconfHook autoPatchelfHook ];
```
Comment (NobbZ) [inline pkgs/development/libraries/ta-lib/default.nix:14]: Are `gcc` and `autoPatchelfHook` really required?

`gcc` should be part of `stdenv` already.

`autoPatchelfHook` shouldn't be needed when building something from source.

File: maintainers/maintainer-list.nix [hunk 2]
```diff
@@ -11524,6 +11524,15 @@
...
   };
+  rafael = {
+    email = "pr9@tuta.io";
+    github = "rafa-dot-el";
+    githubIb = 104688305;
+    keys = [{
+      longkeyid = "ed25519/0x469E3255A40D2AD6";
+      fingerprint = "5F0B 3EAC F1F9 8155 0946 CDF5 469E 3255 A40D 2AD6";
+    }];
+  };
```
Comment (NobbZ) [inline maintainers/maintainer-list.nix:11535]: Can you make this an extra commit, preceeding the one that adds the "package"?

**Review by NobbZ — CHANGES_REQUESTED** [review]
Other than that, it seems to build.

```text
Link to currently reviewing PR:
https://github.com/NixOS/nixpkgs/pull/206363

1 package built:
ta-lib
```

```text
$ result/bin/ta-lib-config --version
0.4.0
```

File: maintainers/maintainer-list.nix (hunk 2 above)
Comment (rafa-dot-el) [reply maintainers/maintainer-list.nix:11535]: Sure, I added it.

File: pkgs/development/libraries/ta-lib/default.nix (hunk 1 above)
Comment (rafa-dot-el) [reply pkgs/development/libraries/ta-lib/default.nix]: I removed then and it builds correctly, thanks for pointing it out!

**Review by NobbZ — APPROVED** [review]

**Pull request merged**